functions.tf

```diff
@@ -176,10 +176,13 @@ locals {
     "WEBSITE_CONTENTSHARE"    = "${local.deployment_container_name}-share"
     "WEBSITE_CONTENTOVERVNET" = 1
     "WEBSITE_VNET_ROUTE_ALL"  = 1
-    "WEBSITE_DNS_SERVER"      = "168.63.129.16"
   }
+  dns_storgage_account_app_settings = {
+    "WEBSITE_DNS_SERVER" = "168.63.129.16"
+  }
+  merged_secured_storage_account_app_settings = var.create_storage_account_private_links ? merge(local.secured_storage_account_app_settings, local.dns_storgage_account_app_settings) : local.secured_storage_account_app_settings
```

Comment:
the condition here should be if we have access to the vnet on the storage account or not
we might not want to create private links because we had them before

Reply:
Maybe we can have just `sa_public_network_access` with 3 options.
like we have in the console and handle it accordingly.